core/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/DateHistogramParser.java

```diff
@@ -52,6 +51,8 @@ protected DateHistogramAggregationBuilder createFactory(String aggregationName,
             factory.interval((Long) interval);
         } else if (interval instanceof DateHistogramInterval) {
             factory.dateHistogramInterval((DateHistogramInterval) interval);
+        } else {
+            throw new AssertionError();
```

Comment:
This is a good point, I am not sure. On the one hand, this is indeed a specific API and restarting is unlikely to fix anything, but on the other hand this is the same for stack overflows are memory leaks?